src/utils/cache.h

```diff
@@ -26,7 +26,10 @@
 
 namespace lczero {
 
-// Generic LRU cache. Thread-safe.
+// Generic LRU cache. Thread-safe. Takes ownership of all values, which are
+// deleted upon eviction; thus, using values stored requires pinning them, which
+// in turn requires Unpin()ing them before cache destruction. The use of
```

Comment:
Well I thought the "before destruction" part is more fundamentally important than the "after use" part, since failing to unpin after use won't cause crashes (maybe it will degrade performance), whereas failing to unpin before destruction obviously does cause crashes. None the less, updated, as this way is perfectly sensible too.